pytorch_lightning/loops/fit_loop.py

```diff
@@ -128,10 +128,10 @@ def restarting(self, restarting: bool) -> None:
         values = (
             self.epoch_progress.current.ready,
             self.epoch_progress.current.started,
-            self.epoch_progress.current.processed,
         )
-        finished_before_on_train_end = any(v != self.epoch_progress.current.completed for v in values)
+        finished_before_on_train_end = any(v != self.epoch_progress.current.processed for v in values)
```

Comment:
in an ideal case when a checkpoint is saved using `ModelCheckpoint`, they are saved inside `on_train_epoch_end`, and `process.current.completed` is not incremented by that time, so during restart, `restarting` will be True and it ends up calling https://github.com/PyTorchLightning/pytorch-lightning/blob/bb81802bff29fa06399d704688df256813e05ad2/pytorch_lightning/loops/fit_loop.py#L271

for the last completed epoch again and then start the next epoch. Due to these certain hooks get called which are not supposed to be called.